ocaml/typing/typecore.ml

```diff
@@ -3604,6 +3605,7 @@ let rec approx_type env sty =
       end
   | Ptyp_arrow (p, arg_sty, sty) ->
       let arg_mode = Typetexp.get_alloc_mode arg_sty in
+      let p = Typetexp.transl_label p in
```

Comment:
Same here. Or maybe the case above isn't a problem, because the type necessarily begins with `'a .` or similar? But then this one is surely a problem.